### ISSUE TITLE
Make HgUtils async

### ISSUE THREAD

**Issue by victor-torres** [victor-torres/atom-hg#15]
spawnSync blocks NodeJS and we can't have another process/thread running it such as a web worker... We have to refactor HgUtils class to use spawn or exec, the asynchronous version of this command. Basically we need to encapsulate some methods into promises and consume them with .then() in HgRepositoryAsync class.

**Comment by victor-torres** [comment]
@mrstebo I further investigated based on your suggestions and I can confirm the blame is in the spawnSync guy.

**Comment by mrstebo** [comment]
I noticed it when I was doing a `console.log` when checking what `hgCommand`. Maybe we should have a separate `HgUtilsAsync` class that can just return promises?

**Comment by victor-torres** [comment]
Yup. Sounds good. Exactly what I would do.

**Comment by victor-torres** [comment]
@mrstebo are you up to this one? May I assign it to you?

**Comment by mrstebo** [comment]
I guess I could have a crack at it 😅

**Comment by mrstebo** [comment]
Yeah. They seemed to have removed the async stuff in Atom now... 
There is an issue [here](https://github.com/atom/atom/issues/12159).
And a discussion [here](https://discuss.atom.io/t/deprecation-of-async-git-in-atom-1-10/31270)
